src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/ExcelCreationServiceForReport.java

```diff
@@ -0,0 +1,240 @@
+    private static final String CLAIMANT_WORK_POSTCODE_HEADER = "Claimant Work Postcode";
+    private static final String RESPONDENT_POSTCODE_HEADER = "Respondent Postcode";
+    private static final String RESPONDENT_ET3_POSTCODE_HEADER = "Respondent ET3 Postcode";
+    private static final List<String> HEADERS = new ArrayList<>(Arrays.asList(
+        CASE_NUMBER_HEADER, DATE_OF_RECEIPT_HEADER, CLAIMANT_POSTCODE_HEADER,
+        CLAIMANT_WORK_POSTCODE_HEADER, RESPONDENT_POSTCODE_HEADER, RESPONDENT_ET3_POSTCODE_HEADER));
+
+    public byte[] getReportExcelFile(ClaimsByHearingVenueReportData reportData, String userName) {
+        if (reportData == null) {
+            return new byte[0];
+        }
+        var reportDetails = reportData.getReportDetails();
+        var workbook = new XSSFWorkbook();
+        var sheet = workbook.createSheet(EXCEL_REPORT_WORKBOOK_NAME);
+        adjustColumnSize(sheet);
+        initializeReportHeaders(reportData, workbook, sheet);
+        initializeReportData(workbook, sheet, reportDetails, userName);
+        return writeExcelFileToByteArray(workbook);
+    }
+
+    private void adjustColumnSize(XSSFSheet sheet) {
+        //Adjust the column width to fit the content
+        for (var i = 0; i <= 5; i++) {
+            sheet.setColumnWidth(i, 9000);
+        }
+    }
+
+    private void initializeReportHeaders(ClaimsByHearingVenueReportData reportData, XSSFWorkbook workbook,
+                                           XSSFSheet sheet) {
+        CellRangeAddress reportTitleCellRange = new CellRangeAddress(0, 0, 0, 6);
+        sheet.addMergedRegion(reportTitleCellRange);
+        XSSFRow rowReportTitle = sheet.createRow(0);
+        rowReportTitle.setHeight((short)(rowReportTitle.getHeight() * 8));
+        CellStyle styleForHeaderCell = getReportTitleCellStyle(workbook);
+        createCell(rowReportTitle, 0, reportData.getDocumentName(), styleForHeaderCell);
+
+        CellRangeAddress reportPeriodCellRange = new CellRangeAddress(1, 1, 0, 6);
+        sheet.addMergedRegion(reportPeriodCellRange);
+        XSSFRow rowReportPeriod = sheet.createRow(1);
+        rowReportPeriod.setHeight((short)(rowReportPeriod.getHeight() * 6));
+        var styleForSubTitleCell = getReportSubTitleCellStyle(workbook);
+        createCell(rowReportPeriod, 0, getReportPeriodAndOfficeName(reportData), styleForSubTitleCell);
+
+        XSSFRow rowHead = sheet.createRow(2);
+        rowHead.setHeight((short)(rowHead.getHeight() * 4));
+        var styleForColHeaderCell = getHeaderCellStyle(workbook);
+        for (var j = 0; j < HEADERS.size(); j++) {
+            rowHead.createCell(j).setCellValue(HEADERS.get(j));
+            createCell(rowHead, j, HEADERS.get(j), styleForColHeaderCell);
+        }
+        createCell(rowHead, HEADERS.size(), "", styleForColHeaderCell);
+    }
+
+    private String getReportPeriodAndOfficeName(ClaimsByHearingVenueReportData reportData) {
```

Comment:
I don't think this method belongs in this class. You're mixing data generation with the excel export/presentation code. Also it is confusing to the reader to add nulls checks when the data should never be null

Reply:
Agreed. Refactoring it.